controller/controller.go

```diff
@@ -62,8 +68,41 @@ const (
 
 	// DefaultIngressThreads is the default number of ingress worker threads to start with the controller
 	DefaultIngressThreads = 10
+
+	// DefaultLeaderElect is the default true leader election should be enabled
+	DefaultLeaderElect = true
```

Comment:
to keep the backward compatibility, default should be false

Reply:
Hi, @harikrongali , thanks for your review. After second thought, I think it is safe to enable leader elect by default. The reasons are
1. `Defaulting to true` is backward compatible. This is because when there is 1 instance, this node will be elected as the master node. So there is no impact on the existing deployment manifest.
2. `Defaulting to true` makes it easier to enable HA because the only change to the install manifest is to increase the replica from 1 to > 1. Otherwise, user has to add `--leader-elect` to the CMD of the manifest.

Does it make sense to you? Thanks.

Reply:
argo-workflows defaults to LeaderElection enabled as well. 

https://github.com/argoproj/argo-workflows/blob/master/docs/environment-variables.md

Looking forward to this getting merged!  Thanks for writing the PR @huikang